### PR TITLE
Parse additional items from history stream

### PR DESCRIPTION
**Pull request by pborzenkov**
Parse additional items from the history stream to be able to calculate
statistics such as:
  - average distance per ride
  - the most popular bike id
  - average ride speed

Signed-off-by: Pavel Borzenkov <pavel.borzenkov@gmail.com>

### CODE REVIEW

File: velobike/history.go [hunk 1]
```diff
@@ -17,10 +17,15 @@ type HistoryItem struct {
...
-	StartDate              *string  `json:"StartDate,omitempty"`
+	StartDate              *string  `json:"StartDate,omitempty"`              // layout: 2006-01-02T15:04:05
 	StartBikeParkingNumber *string  `json:"StartBikeParkingNumber,omitempty"` // Only for type "Ride"
 	EndBikeParkingNumber   *string  `json:"EndBikeParkingNumber,omitempty"`   // Only for type "Ride"
 	Time                   *string  `json:"Time,omitempty"`                   // Only for type "Ride"
+	BikeID                 *string  `json:"BikeId,omitempty"`                 // Only for type "Ride"
+	EndDate                *string  `json:"EndDate,omitempty"`                // Only for type "Ride", layout: 2006-01-02T15:04:05
+	Distance               *int     `json:"CoveredDistance,omitempty"`        // Only for type "Ride", in meters
+	// Duration is always 5-10 seconds less than Time. I suspect it represents Time minus time needed to lock/unlock the bike.
+	Duration *string `json:"Duration"`
```
Comment (rumyantseva) [inline velobike/history.go:28]: What kind of data `Duration` contains? Why did you choose string type? 

May you show, please, an example response? Unfortunately, I don't ride this year, so I can't check it myself.

**Comment by rumyantseva** [comment]
Thank you for the improvement! 
Looks like there are a lot of new features in velobike this year :)

File: velobike/history.go (hunk 1 above)
Comment (pborzenkov) [reply velobike/history.go:28]: Duration is in the same format as Time:

```
        {                              
            "BikeId": "1756",          
            "BikeType": "ordinary",    
            "CoveredDistance": 3038,   
            "Duration": "00.00:17:34", 
            "EndBikeParkingAddress": "469 - Территория ВДНХ (у прудов)",       
            "EndBikeParkingName": "Проспект Мира, 119с353 (ВДНХ)",             
            "EndBikeParkingNumber": "0469",                                    
            "EndBikeSlotNumber": "1",  
            "EndDate": "2017-05-13T14:04:04",                                  
            "Id": "0",                 
            "Price": 0.0,              
            "Rejected": false,         
            "StartBikeParkingAddress": "481 - ул. Милашенкова, д.12Д",         
            "StartBikeParkingName": "улица Милашенкова, д.12",                 
            "StartBikeParkingNumber": "0481",                                  
            "StartBikeSlotNumber": "", 
            "StartDate": "2017-05-13T13:46:26",                                
            "Text": "Я проехал от станции 0481 до 0469.",                      
            "Time": "00.00:17:38",     
            "Type": "Ride"             
        },
```

**Comment by pborzenkov** [comment]
I haven't added all the available fields with this patch, as I don't need them right now. But I can do this for the sake of completeness.

File: velobike/history.go (hunk 1 above)
Comment (rumyantseva) [reply velobike/history.go:28]: Ok, I see! Thank you. Let's use `string` for this major version, yeah.

Comment (pborzenkov) [reply velobike/history.go:28]: Yeah, ideally, for the next major version it'd better to have a custom json unmarshaler that will convert Time and Duration to time.Duration type.

Comment (rumyantseva) [reply velobike/history.go:28]: Created an issue #5 about it.

**Pull request merged**